plugins/tedge_apama_plugin/src/main.rs

```diff
@@ -48,128 +48,244 @@ pub enum PluginOp {
-            );
-            fs::create_dir_all(tedge_apama_project_path)?;
-            fs::rename(tmp_apama_project_path, tedge_apama_project_path)?;
-            println!("Installation of new project successful");
+fn install_project(project_archive_path: &Path) -> Result<(), InternalError> {
+    let tedge_apama_project_path: &Path = Path::new(TEDGE_APAMA_PROJECT_DIR);
+    let tmp_apama_project_path: &Path = Path::new(TMP_APAMA_PROJECT_DIR);
 
-            println!("Restarting apama to load the new project");
-            run_cmd("service", "apama restart")?;
-            println!("Restart of apama service successful");
+    let archive_file = File::open(&project_archive_path)?;
 
-            success
-        }
+    let mut archive = zip::ZipArchive::new(archive_file)?;
 
-        PluginOp::Remove {
-            module: _,
-            version: _,
-        } => {
-            if tedge_apama_project_path.exists() {
-                println!("Stopping apama service");
-                run_cmd("service", "apama stop")?;
-                println!("Stopping apama service successful");
+    if let Err(zip::result::ZipError::FileNotFound) = archive.by_name("project/") {
+        return Err(InternalError::InvalidProjectArchive);
+    }
 
-                println!("Removing existing project at {}", TEDGE_APAMA_PROJECT_DIR);
-                fs::remove_dir_all(tedge_apama_project_path)?;
-                println!("Removal of existing project successful");
-            }
+    println!("Extracting the archive at {:?}", project_archive_path);
+    archive.extract(tmp_apama_project_path)?;
+    println!("Extraction successful");
 
-            success
-        }
-    };
+    // Deleting existing project as the rename API expects the target dir to be empty
+    if tedge_apama_project_path.exists() {
+        println!("Removing existing project at {}", TEDGE_APAMA_PROJECT_DIR);
+        fs::remove_dir_all(tedge_apama_project_path)?;
+        println!("Removal of existing project successful");
+    }
+
+    println!(
+        "Installing newly extracted project to {}",
+        TEDGE_APAMA_PROJECT_DIR
+    );
+    fs::create_dir_all(tedge_apama_project_path)?;
+    fs::rename(
+        tmp_apama_project_path.join("project"),
+        tedge_apama_project_path,
+    )?;
+    println!("Installation of new project successful");
+
+    println!("Restarting apama to load the new project");
+    let status = run_cmd("service", "apama restart")?;
+    println!("Restart of apama service successful");
 
     Ok(status)
 }
 
-fn run_cmd(cmd: &str, args: &str) -> Result<ExitStatus, InternalError> {
+fn remove_project(_project_name: &str) -> Result<(), InternalError> {
+    let tedge_apama_project_path: &Path = Path::new(TEDGE_APAMA_PROJECT_DIR);
+
+    if tedge_apama_project_path.exists() {
+        println!("Stopping apama service");
+        run_cmd("service", "apama stop")?;
+        println!("Stopping apama service successful");
```

Comment:
This pattern occurs in many places. It might be good to add functions that start/stop/restart apama adding the extra logging infos.